### PR TITLE
Null exception setting email

### PR DESCRIPTION
**Pull request by andrespelaezp**
RemoteParams null validation was added once again, also a state for the dialog to be closed when the Set Email task is scheduled.

## 
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-android-sdk/1168)

### CODE REVIEW

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java
Comment (Jeasmine) [inline OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1325]: remote params shouldn't be null at this point, because of the shouldQueueTaskForInit check. Why is it needed?

Comment (andrespelaezp) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1325]: When there is a connection issue and the remote configuration is not previously setup, then `getRemoteParams()` is Null and this is the breaking point for the issue found.

Comment (Jeasmine) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1325]: Sry I might be missing some part, but shouldQueueTaskForInit check for remote == null, if remote is still null because the network failed, then the task should be queue again, or why is it not being queue again?

Comment (andrespelaezp) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1325]: It does get queued, but it also continues evaluating the statements below. Another approach could be a return statement within the task schedule block, what do you think?

Comment (Jeasmine) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1325]: oooh I see, that method is missing the return! good catch, so that is why is failing

Comment (Jeasmine) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1325]: No other changes should be needed but the return stmt

Comment (Jeasmine) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1325]: compare with the other methods using shouldQueueTaskForInit

**Review by Jeasmine — COMMENTED** [review]
The last commit is LGTM, we should agree if we want the user to know if the task got queue.
`void onTaskQueued();`
From my point of view, that is something the user doesn't have to know, the only information he wants to know if success and failure. Let me know what you think, also @jkasten2 thoughts?

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java
Comment (jkasten2) [inline OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1304]: I don't think we should change / add new APIs to address this bug fix PR. However I am open to a proposal on a new API to best handle this scenario for all our methods for consistency.

Just some feedback on the API itself though, I would expect a return value from `setEmail` instead of a callback that it has been queued.

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java
Comment (jkasten2) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1326]: This is redundant code, the `shouldQueueTaskForInit` check and the `return` added above would cover this so I think we should remove the `null` check that was added in this PR.

**Review by jkasten2 — CHANGES_REQUESTED** [review]

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java
Comment (andrespelaezp) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1304]: Noted, I will keep this in mind for a future integration

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java
Comment (jkasten2) [inline OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java:1641]: This also needs to be removed

Comment (jkasten2) [inline OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java:1641]: TravisCI is failing due to a Java compile error due to this interface method no longer existing.

**Review by jkasten2 — CHANGES_REQUESTED** [review]

**Review by jkasten2 — APPROVED** [review]
Looks good to me. I recommend we squash all commit down to one before merging however.

**Pull request merged**
